### ISSUE TITLE
[Enhancement] Single Point of Build

### ISSUE THREAD

**Issue by adrianriobo** [containers/podman-desktop-e2e#168]
The idea is to avoid duplication and over computation across the podman desktop ecosystem, each extension to be tested requires the latest build of Podman Desktop built for development. 

As so instead of add that building process across each pipeline/flow on each platform / component there will be one flow doing it on each merge on main branch for Podman Desktop that will send an event to a webhook on [podman-desktop-e2e](https://github.com/containers/podman-desktop-e2e) which will trigger a build of the development version and packetize it to be then reused from any other flow / pipeline

![podman-desktop-testing drawio](https://github.com/containers/podman-desktop-e2e/assets/1957899/d579f121-815a-4c27-9088-34f3e46aa636)

**Comment by adrianriobo** [comment]
Side note if use deliverest as base image for them we will have something to directly copy the development version to a target host

**Comment by adrianriobo** [comment]
Should we keep the same version of PD during a extension PR life cycle @benoitf  (i.e even if we manage to use pre-release versions of PD, should we lock the PR pre release version to run the PR until it got merged?)

**Comment by benoitf** [comment]
@adrianriobo 

how would it be possible to change the version of Podman Desktop during the test of an extension PR ?

you mean, if someone add a new commit later to the PR of an extension, maybe during that time Podman Desktop had new commits ?

I would assume that each new run (after commit/amend) on the extension PR will fetch the current main build of Podman Desktop

**Comment by adrianriobo** [comment]
Yeah exactly; lets imagine I create a PR on a extension today and it is tested against the current pre-released version of PD on gh; them tomorrow a new commit (or force push) comes into the extension PR and the pipeline is triggered, by then maybe there is a different latest pre- release vearion of PD on gh.... as so the PR will be tested against a different PD version.

> I would assume that each new run (after commit/amend) on the extension PR will fetch the current main build of Podman Desktop

I think the same, there should not be an issue having different PD versions on each PR run, as code should not be dependent
